### PR TITLE
feat: upgrade to palm

### PR DESCRIPTION
**Pull request by Faraz32123**
v16.0.0

### CODE REVIEW

File: tutorecommerce/plugin.py [hunk 1]
```diff
@@ -180,3 +183,35 @@
 tutor_hooks.Filters.CONFIG_OVERRIDES.add_items(
     list(config.get("overrides", {}).items())
 )
+
+
+@tutor_hooks.Filters.APP_PUBLIC_HOSTS.add()
+def _print_ecommerce_public_hosts(hosts: list[str], context_name: t.Literal["local", "dev"]) -> list[str]:
+    if context_name == "dev":
+        hosts += ["ecommerce.{{ LMS_HOST }}:8130"]
```
Comment (regisb) [inline tutorecommerce/plugin.py]: Replace "ecommerce.{{ LMS_HOST }}" by "{{ ECOMMERCE_HOST }}" (on this line and the next one)

File: tutorecommerce/templates/ecommerce/apps/ecommerce/settings/partials/common.py [hunk 2]
```diff
@@ -91,6 +91,8 @@
 PAYMENT_PROCESSOR_CONFIG = {
     "openedx": common_payment_processor_config,
     "dev": common_payment_processor_config,
+    "edx" : common_payment_processor_config,
+    "other" : common_payment_processor_config,
```
Comment (regisb) [inline tutorecommerce/templates/ecommerce/apps/ecommerce/settings/partials/common.py:95]: Why did you add these new payment processors?

**Review by regisb — COMMENTED** [review]
Thanks a lot for this PR! Much appreciated.

File: tutorecommerce/templates/ecommerce/apps/ecommerce/settings/partials/common.py (hunk 2 above)
Comment (Faraz32123) [reply tutorecommerce/templates/ecommerce/apps/ecommerce/settings/partials/common.py:95]: Screenshot 2023-06-01 at 3 19 12 PM

I was getting an error due to no key "edx".

**Comment by regisb** [comment]
When I attempt to build all images, the command fails with the following error:

```
$ tutor images build all
...
Building image docker.io/overhangio/openedx-orders-dev:16.0.0
docker buildx build --tag=docker.io/overhangio/openedx-orders-dev:16.0.0 --output=type=docker --target=orders-dev --cache-from=type=registry,ref=docker.io/overhangio/openedx-orders-dev:16.0.0-cache /home/tutor/.local/share/tutor/env/plugins/mfe/build/mfe
[+] Building 0.0s (2/2) FINISHED                                                                                                                                                                           
 => [internal] load build definition from Dockerfile                                                                                                                                                  0.0s
 => => transferring dockerfile: 16.73kB                                                                                                                                                               0.0s
 => [internal] load .dockerignore                                                                                                                                                                     0.0s
 => => transferring context: 2B                                                                                                                                                                       0.0s
ERROR: failed to solve: target stage orders-dev could not be found
Error: Command failed with status 1: docker buildx build --tag=docker.io/overhangio/openedx-orders-dev:16.0.0 --output=type=docker --target=orders-dev --cache-from=type=registry,ref=docker.io/overhangio/openedx-orders-dev:16.0.0-cache /home/tutor/.local/share/tutor/env/plugins/mfe/build/mfe
```

Can you please investigate?

**Comment by Faraz32123** [comment]
Sure thing, I can investigate.

File: tutorecommerce/plugin.py [hunk 3]
```diff
@@ -78,6 +82,27 @@
...
+    },
+}
+
+
+# The core MFEs are added with a high priority, such that other users can override or
+# remove them.
+@MFE_APPS.add(priority=tutor_hooks.priorities.HIGH)
+def _add_core_mfe_apps(apps: dict[str, MFE_ATTRS_TYPE]) -> dict[str, MFE_ATTRS_TYPE]:
+    apps.update(CORE_MFE_APPS)
+    return apps
```
Comment (regisb) [inline tutorecommerce/plugin.py:100]: Let's not add ecommerce MFE apps with a high priority. This should only be done for core MFE apps. Similarly, let's not create a CORE_MFE_APPS, which is an ambiguous name. Instead, I suggest:

```
@MFE_APPS.add()
def _add_ecommerce_mfe_apps(apps: dict[str, MFE_ATTRS_TYPE]) -> dict[str, MFE_ATTRS_TYPE]:
    apps.update({
        "orders": {
            "repository": "https://github.com/edx/frontend-app-ecommerce",
            "port": 1996,
        },
        "payment": {
            "repository": "https://github.com/edx/frontend-app-payment",
            "port": 1998,
        },
    })
    return apps
```

**Comment by regisb** [comment]
It looks like this plugin is broken when we upgrade to node 18... https://github.com/openedx/wg-build-test-release/issues/261#issuecomment-1584746372

**Comment by regisb** [comment]
Merging to palm, let's discuss [there](https://github.com/overhangio/tutor-ecommerce/pull/42).
